test/NonSiloTests/Orleans.NonSiloTests.csproj

```diff
@@ -48,21 +48,43 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
+    <Compile Include="AssemblyLoaderTests.cs" />
+    <Compile Include="AsynchAgentRestartTest.cs" />
+    <Compile Include="AsyncPipelineTests.cs" />
+    <Compile Include="Async_AsyncExecutorWithRetriesTests.cs" />
+    <Compile Include="Async_TimingTests.cs" />
+    <Compile Include="CodeGeneratorTests.cs" />
     <Compile Include="ConfigTests.cs" />
+    <Compile Include="CounterTests.cs" />
+    <Compile Include="ErrorInjectionStorageProvider.cs" />
+    <Compile Include="General\CounterStatisticTests.cs" />
+    <Compile Include="General\Identifiertests.cs" />
+    <Compile Include="General\InterfaceRules.cs" />
+    <Compile Include="General\InternalSerializationTests.cs" />
+    <Compile Include="General\LruTest.cs" />
+    <Compile Include="General\RequestContextTestsNonSiloRequired.cs" />
+    <Compile Include="General\RingTests_Standalone.cs" />
+    <Compile Include="ManagementAgentTests.cs" />
+    <Compile Include="OrleansRuntime\AsyncSerialExecutorTests.cs" />
+    <Compile Include="OrleansRuntime\ExceptionsTests.cs" />
+    <Compile Include="OrleansRuntime\Streams\BestFitBalancerTests.cs" />
+    <Compile Include="OrleansRuntime\Streams\SubscriptionMarkerTests.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
     <Compile Include="..\..\src\Build\GlobalAssemblyInfo.cs">
       <Link>Properties\GlobalAssemblyInfo.cs</Link>
     </Compile>
     <Compile Include="SchedulerTests\OrleansTaskSchedulerAdvancedTests.cs" />
     <Compile Include="SchedulerTests\OrleansTaskSchedulerAdvancedTests_Set2.cs" />
     <Compile Include="SchedulerTests\OrleansTaskSchedulerBasicTests.cs" />
-    <Compile Include="SerializationTests\BuiltInSerializerTests.cs" />
+    <Compile Include="SerializationTests\BondSerializationTests.cs" />
```

Comment:
I'm not 100% convinced that we should have separate project for providers.  We can give it tags or in its own folder if you think you think it's hard to find all the tests related to external providers. 

The thing is, if we categorize too much, it would be hard to find a place to put tests like BondSerializationTests or test like GatewaySelectionTest, where only one test in those file related to SqlMembershipTable. So in this situation , we would split the tests into two files and one in NonSiloTests, one in  SQLMemebershipTests. It's confusing to find all the tests related to GatewaySeleciton in this case